assignmen_week1.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+)
+
+func parse(input string) (a float64, b float64, op string, err error) {
+	params := strings.Fields(input)
+
+	if len(params) < 3 {
+		return 0.0, 0.0, "", errors.New("Error: Invalid input")
+	}
+
+	x, err1 := strconv.ParseFloat(params[0], 64)
+	if err1 != nil {
+		return 0.0, 0.0, "", err1
```

Comment:
Just `return` is good enough if you change `err1` to `err`, actually we don't need to use different variables for storing the error, only one is good enough.